prover/src/constraints/evaluator.rs

```diff
@@ -148,7 +148,7 @@ impl<'a, A: Air, E: FieldElement<BaseField = A::BaseField>> ConstraintEvaluator<
         fragment: &mut EvaluationTableFragment<E>,
     ) {
         // initialize buffers to hold trace values and evaluation results at each step;
-        let mut main_frame = EvaluationFrame::new(trace.main_trace_width());
+        let mut main_frame = A::Frame::new(trace.main_trace_width());
```

Comment:
Can we move the EvaluationFrame constructors outside of the trait definition while still initializing a buffer outside of the loop below?

Reply:
We could put a method for building "blank" frame into the `Air` trait (so, `Air` trait becomes kind of like a factory for evaluation frames). But I'm not seeing why having a constructor on the frame itself is an issue.

One thing I'd potentially do differently is to change the signature of the constructor to take a reference an instance of an `Air` struct. For example, something like:
```Rust
fn new<A: Air<BaseField = Self::BaseField>>(air: &A) -> Self;
```

Reply:
We could put a method for building "blank" frame into the `Air` trait (so, `Air` trait becomes kind of like a factory for evaluation frames). But I'm not seeing why having a constructor on the frame itself is an issue.

One thing I'd potentially do differently is to change the signature of the constructor to take a reference an instance of an `Air` struct. For example, something like:
```Rust
fn new<A: Air<BaseField = Self::BaseField>>(air: &A) -> Self;
```

Reply:
The motivation was to allow each struct to potentially customize the arguments needed for its constructor. I think your suggestion solves that by giving access to the `AirContext`, which should be all the information the constructor needs.